client/reader/sidebar/expandable-add-form.jsx

```diff
@@ -1,45 +1,35 @@
 /**
  * External Dependencies
  */
-import React from 'react';
+import React, { Component, PropTypes } from 'react';
 import classNames from 'classnames';
-import noop from 'lodash/noop';
+import { noop, identity } from 'lodash';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal Dependencies
  */
 import Gridicon from 'components/gridicon';
 import Button from 'components/button';
 
-const ExpandableSidebarAddForm = React.createClass( {
+export class ExpandableSidebarAddForm extends Component {
 
-	propTypes: {
-		addLabel: React.PropTypes.string,
-		addPlaceholder: React.PropTypes.string,
-		onAddSubmit: React.PropTypes.func,
-		onAddClick: React.PropTypes.func
-	},
-
-	getInitialState() {
-		return {
-			isAdding: false
-		};
-	},
-
-	getDefaultProps() {
-		return {
-			onAddSubmit: noop,
-			onAddClick: noop
+	constructor() {
+		super();
+		this.toggleAdd = this.toggleAdd.bind( this );
+		this.handleAddKeyDown = this.handleAddKeyDown.bind( this );
+		this.state = {
+			isAdding: false,
 		};
-	},
-
+	}
+	
 	toggleAdd() {
```

Comment:
Not a big deal at all, tomaytoes/tomahtoes, but we can do `toggleAdd = () =>` now to get the binding to `this`